### ISSUE TITLE
Opening a project in the GUI loads all elements and data at the same time, causes high network load

### ISSUE THREAD

**Issue by ThunderRush** [rancher/rancher#25691]
**What kind of request is this (question/bug/enhancement/feature request):**
Enchancement (See first comment)

**Steps to reproduce (least amount of steps as possible):**
Set up Rancher HA on RKE Cluster behind Loadbalancer
Deploy cluster
Open System project

**Result:**
Opening the project on the GUI causes a massive network spike lasting a few minutes, coming from the rancher-cluster-agent pod, to the loadbalancer, and then to the Rancher HA cluster.

**Other details that may be helpful:**
This only happens in one of our clusters, for some reason. Besides triggering it manually it also occurs every 30 minutes,

I have tried looking around in the logfiles on the node that hosts the pod, as well as the pod itself. There is absolutely nothing that would point to it causing such high traffic.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.8
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): vSphere
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.9", GitCommit:"3e4f6a92de5f259ef313ad876bb008897f6a98f0", GitTreeState:"clean", BuildDate:"2019-08-05T09:22:00Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:16Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.06.3-ce
 API version:       1.38
 Go version:        go1.10.4
 Git commit:        d7080c1
 Built:             Wed Feb 20 02:24:22 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.3-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       d7080c1
  Built:            Wed Feb 20 02:25:33 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

**Comment by ThunderRush** [comment]
Alright, we have figured out the solution to the issue, but now it's really more of an enhancement.

The issue was caused by a large number of configmaps in the kube-system namespace. We did not set a limit on our helm revisions, and one of our developers deployed quite a bit and quite a lot. Every deployment created a 400kb configmap - and we had over 700 of them at one point.

We have solved this issue by simply limiting the max-history with helm, but this really causes another question: when opening a project, is Rancher actually loading every single configmap/deployment/etc. to present to the user as opposed to when requesting it?

We only noticed this issue because Rancher is running behind an F5 Loadbalancer with a license maximum of 200mbit/s. It still caused a lot of traffic, and I don't think this is good/intended behavior.  Cluster resources in a project should really only be loaded if a user actually requests them by interacting with it.
